lua/wire/stools/ledtape.lua

```diff
@@ -0,0 +1,186 @@
+WireToolSetup.open( "ledtape", "LED Tape", "gmod_wire_ledtape", nil, "LED Tape Controllers" )
+
+--[[
+	I would have used the original wirepath system, but it looks like it was never designed for a tool like this.  
+	So we're stuck with whatever horribleness I come up with.  Sorry.
+	- Fast
+]]--
+
+TOOL.ClientConVar = {
+	material = "cable/white",
+	width = "1",
+	model = "models/beer/wiremod/hydraulic.mdl",
+	modelsize = ""
+}
+
+TOOL.ToolPath = {}
+TOOL.CurWidth = 1
+TOOL.Scale = 1
+
+function TOOL:GetConVars()
+	return self:GetClientInfo("material") or "cable/rope",
+		   self:GetClientNumber("width", 3)
+end
+
+local function isLookingAtController(trace)
+	return trace.Entity and trace.Entity:GetClass() == "gmod_wire_ledtape"
+end
+
+if CLIENT then
+
+	language.Add( "Tool.wire_ledtape.name", "LED Tape Tool (Wire)" )
+	language.Add( "Tool.wire_ledtape.desc", "Makes a group of ropes with controllable color" )
+	language.Add( "Tool.wire_ledtape.width", "Width:" )
+	language.Add( "Tool.wire_ledtape.material", "Material:" )
+	TOOL.Information = {
+		{ name = "right_0", stage = 0, text = "Start LED Tape" },
+		{ name = "left_0", stage = 0, text = "Update material of existing controller" },
+		{ name = "right_1", stage = 1, text = "Place another point" },
+		{ name = "left_2", stage = 2, text = "Finish tape and place controller" },
+		{ name = "right_2", stage = 2, text = "Place more points" },
+		{ name = "left_3", stage = 3, text = "Finish tape and place controller" },
+	}
+	WireToolSetup.setToolMenuIcon( "icon16/chart_line.png" )
+
+	local WHITE = Color(255,255,255) -- creating these every frame is bad
+	local YELLOW = Color(255,255,0)  -- it wastes memory and strains the garbage collector
+
+	function TOOL:Holster()
+		self.ToolPath = {}
+		self:ReleaseGhostEntity()
+	end
+
+	function TOOL:Preview()
+
+		if #self.ToolPath < 1 then hook.Remove("PostDrawOpaqueRenderables","LEDTape_Preview") return end -- something happened, bail
+
+		render.SetMaterial(self.CurMater)
+
+		local pt2 = Wire_LEDTape.DrawFullbright(self.CurWidth, self.ScrollMul / 3, WHITE, self.CurMater, self.ToolPath)
+
+		if pt2 and (#self.ToolPath + 1 < Wire_LEDTape.MaxPoints) then
+			local eyetrace = LocalPlayer():GetEyeTrace()
+			local pt3 = eyetrace.HitPos + eyetrace.HitNormal * self.CurWidth * 0.5
+			render.DrawLine( pt2, pt3, YELLOW )
+		end
+
+	end
+
+end
+
+WireToolSetup.BaseLang()
+WireToolSetup.SetupMax(8)
+
+if SERVER then
+
+	function TOOL:MakeEnt(ply, model, Ang, trace)
+		local material, width = self:GetConVars()
+		return MakeWireLEDTapeController(ply, trace.HitPos, Ang, model, self.ToolPath, width, material)
+	end
+
+	function TOOL:Holster()
+		self:SetStage(0)
+		self.ToolPath = {}
+	end
+
+end
+
+function TOOL:RightClick( trace )
+
+	if not trace.Hit or ( trace.Entity:IsValid() and trace.Entity:IsPlayer() ) or trace.Entity:IsWorld() then return end
+	if ( SERVER and not util.IsValidPhysicsObject( trace.Entity, trace.PhysicsBone ) ) then return false end
```

Comment:
Is there any functional reason it's restricted like this?

Reply:
Due to the way the beams are parented to what they attach to, this is unfortunately necessary.  Regular wires (which is what this code is based on) have the same restriction.

Reply:
Alright, thanks, just didn't know that. Thought it was like ropes.